diff --git a/clang/lib/Driver/ToolChains/Gnu.cpp b/clang/lib/Driver/ToolChains/Gnu.cpp
index e5e1b1d77269..8bd7399ca298 100644
--- a/clang/lib/Driver/ToolChains/Gnu.cpp
+++ b/clang/lib/Driver/ToolChains/Gnu.cpp
@@ -517,6 +517,18 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
 
   Args.addAllArgs(CmdArgs, {options::OPT_L, options::OPT_u});
 
+  // prefer bundled libs over those from other sources
+  if(isa<LinkJobAction>(JA) || isa<LinkerWrapperJobAction>(JA)) {
+    const char *Exec = D.getClangProgramPath();
+    SmallString<256> LibArg("-L");
+    auto InstallDir = llvm::sys::path::parent_path(
+                      llvm::sys::path::parent_path(Exec));
+    LibArg += InstallDir;
+    LibArg += "/";
+    LibArg += CLANG_INSTALL_LIBDIR_BASENAME;
+    CmdArgs.push_back(Args.MakeArgStringRef(LibArg));
+  }
+
   ToolChain.AddFilePathLibArgs(Args, CmdArgs);
 
   if (D.isUsingLTO()) {
